src/odb/src/db/dbNet.cpp

```diff
@@ -727,6 +727,30 @@ bool dbNet::rename(const char* name)
   return true;
 }
 
+void dbNet::swapNetNames(dbNet* source)
+{
+  _dbNet* dest_net = (_dbNet*) this;
+  _dbNet* source_net = (_dbNet*) source;
+  _dbBlock* block = (_dbBlock*) source_net->getOwner();
+
+  std::string source_name = source_net->_name;
+  std::string dest_name = dest_net->_name;
+
+  block->_net_hash.remove(dest_net);
+  free((void*) dest_net->_name);
+
+  block->_net_hash.remove(source_net);
+  free((void*) source_net->_name);
+
+  dest_net->_name = strdup(source_name.c_str());
+  ZALLOCATED(dest_net->_name);
+  block->_net_hash.insert(dest_net);
+
+  source_net->_name = strdup(dest_name.c_str());
+  ZALLOCATED(source_net->_name);
+  block->_net_hash.insert(source_net);
```

Comment:
Why not remove from both hashes,  just swap the names rather the deallocating and reallocating them, and reinsert?

Reply:
Because (1) The name is annotated with zallocated (even if this is not used, it shows the intent to lock the memory to a location, which means it cannot be simply swapped but needs to be reallocated). (b) The hash insert references the name.

Reply:
On looking at the zallocated macro in OpenROAD I can see it has nothing to do with zalloc (zone allocation which is used by those familiar with memory management in embeddes systems) and the intent was just to test for a null pointer for a name, so doubtless this can be swapped without reallocation.

Reply:
So in the latest commit this has the pointer swapping as opposed to reallocation and swapping.